index.js

```diff
@@ -158,9 +177,14 @@ function parse (opts, cb) {
     if (!fs.existsSync(file) || fs.statSync(file).isDirectory()) {
       var filename = './' + path.basename(file)
       debug('resolve', [path.dirname(file), filename])
-      file = resolve.sync(filename, { basedir: path.dirname(file), extensions: extensions })
+      file = resolve.sync(filename, { basedir: path.dirname(file), extensions: Object.keys(extensions) })
     }
 
+    var ext = path.extname(file)
+    var detective = extensions[ext]
+
+    if (typeof detective !== 'function') return callback(new Error('Detective function missing for "' + ext + '"'))
```

Comment:
It's the same behaviour, slightly different error message. I can keep the existing error message, but it's unclear which extension is failing for the existing error message.